### ISSUE TITLE
AI Chat: Fix same-document navigations

### ISSUE THREAD

**Issue by petemill** [brave/brave-browser#34946]
Go to:
https://github.com/brave/brave-browser/issues
2. Open Leo. Notice the panel shows the page context is for brave-browser/issues
3. Click on an issue and the page content changes
4. Check the Leo pane again. It doesn't update.
If you repeat these steps by clicking on a link on another page though it works.

**Comment by kjozwiak** [comment]
Adding `QA/Blocked` as the above will still need to be uplifted into `1.62.x` once `C121` is merged into `1.62.x`.

**Comment by kjozwiak** [comment]
The above requires `1.62.149` or higher for `1.62.x` verification 👍

**Comment by stephendonner** [comment]
## Verification `PENDING` using

```
Brave | 1.62.149 Chromium: 121.0.6167.75 (Official Build) (x86_64)
-- | --
Revision | ff84587bd70af9fcbcbe59fc5194ca65082759c4
OS | macOS Version 11.7.10 (Build 20G1427)
```


### Steps:
1. installed `1.62.149`
2. launched Brave using `--env-leo=staging --env-ai-chat.bsg=dev --env-ai-chat-premium.bsg=dev --enable-logging=stderr --v=2`
3. loaded github.com
4. clicked on Leo in the sidebar
5. clicked on Summarize this page
6. clicked on Accept and begin
7. looked at the page-context header
8. confirmed it read Issues - brave/brave-browser - GitHub
9. clicked on the 1st issue present
10. confirmed it read Instagram not showing any images only reels - GitHub

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using
```
Brave | 1.62.149 Chromium: 121.0.6167.75 (Official Build) (64-bit)
-- | --
Revision | ff84587bd70af9fcbcbe59fc5194ca65082759c4
OS | Windows 11 Version 22H2 (Build 22621.3007)
```

1. Installed 1.62.159
2. launched Brave
3. visited https://github.com/brave/brave-browser/issues
4. clicked sidebar on the toolbar 
5. clicked Leo icon
6. clicked `Summarize the page`  
7. clicked `Accept and begin`
8. verified Leo panel shows summary for the loaded page as expected
9. clicked on an issue (first issue) 

#### Confirmed Leo panel is updated/loaded to show the template and previous page summary is no longer shown


1.63.126 (before change) 
example | example 
--------| ------------
<img width="960" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/0a4db03a-5035-4f52-b568-88ca84c797e9">|<img width="955" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/003dbba3-361a-4859-8fcc-09c625067620">



1.62.149
example | example  
----- | ----- 
<img width="956" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/e97e0340-ef17-48b6-a236-911b1fa7bf6c">|<img width="956" alt="image" src="https://github.com/brave/brave-browser/assets/98358127/83aebe08-10e7-4cc5-88f6-949893406f01">
